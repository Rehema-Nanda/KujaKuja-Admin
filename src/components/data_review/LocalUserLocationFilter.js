import React, { Component } from "react";
import {
    Popover, PopoverBody, Input, Label, FormGroup, Badge,
} from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

class LocalUserLocationFilter extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isTeamLocationDropdownOpen: false,
        };
    }

    truncate = (str) => {
        const length = 14;

        if (str.length <= length) {
            return str;
        }

        return `${str.substring(0, length)}...`;
    };

    toggleTeamLocationDropdown = () => {
        const { isTeamLocationDropdownOpen } = this.state;

        this.setState({
            isTeamLocationDropdownOpen: !isTeamLocationDropdownOpen,
        });
    };

    getSelectedTeamLocationName = () => {
        const { locations, selectedUserLocations } = this.props;

        if (selectedUserLocations.length === 1) {
            const location = locations.find((loc) => {
                return selectedUserLocations[0] === loc.id;
            });

            if (location) {
                return this.truncate(location.name);
            }
        }

        return "Team Location";
    };

    render() {
        const { isTeamLocationDropdownOpen } = this.state;

        const { locations, handleUserLocationChange, selectedUserLocations } = this.props;

        return (
            <div className="user-location-container">
                <div
                    id="UserLocationDropdown"
                    className="map-controls-button user-location-button"
                    onClick={this.toggleTeamLocationDropdown}
                >
                    <Badge color="secondary" pill>
                        {selectedUserLocations.length > 0 ? selectedUserLocations.length : "All"}
                    </Badge>
                    {" "}
                    {this.getSelectedTeamLocationName()}
                    <div className="map-controls-button-caret">
                        <FontAwesomeIcon icon="caret-down" />
                    </div>
                </div>
                <Popover
                    placement="bottom"
                    isOpen={isTeamLocationDropdownOpen}
                    target="UserLocationDropdown"
                    toggle={this.toggleTeamLocationDropdown}
                    className="map-controls-popover"
                >
                    <PopoverBody>
                        <ul className="map-controls-popover-list">
                            {locations.map((location) => {
                                return (
                                    <li key={location.id}>
                                        <FormGroup check>
                                            <Label check className="map-controls-checkbox">
                                                <Input
                                                    type="checkbox"
                                                    onChange={handleUserLocationChange}
                                                    value={location.id}
                                                    checked={selectedUserLocations.includes(location.id)}
                                                />
                                                {location.name}
                                            </Label>
                                        </FormGroup>
                                    </li>
                                );
                            })}
                        </ul>
                    </PopoverBody>
                </Popover>
            </div>
        );
    }
}

export default LocalUserLocationFilter;
