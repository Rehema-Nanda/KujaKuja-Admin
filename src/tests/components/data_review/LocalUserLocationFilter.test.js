import React from "react";
import { mount } from "enzyme";

import LocalUserLocationFilter from "../../../components/data_review/LocalUserLocationFilter";

describe("LocalUserLocationFilter component", () => {
    const testProps = {
        locations: [
            { id: "1", name: "Test Location" },
            { id: "2", name: "Test Long Location" },
        ],
        handleUserLocationChange: jest.fn(),
        selectedUserLocations: [],
    };

    const wrapper = mount(<LocalUserLocationFilter {...testProps} />);

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should not truncate long strings", () => {
        const props = {
            ...testProps,
            selectedUserLocations: ["1"],
        };
        const tree = mount(<LocalUserLocationFilter {...props} />);
        const btn = tree.find(".user-location-button");
        expect(btn.text()).toBe("1 Test Location");
    });

    it("should truncate long strings", () => {
        const props = {
            ...testProps,
            selectedUserLocations: ["2"],
        };
        const tree = mount(<LocalUserLocationFilter {...props} />);
        const btn = tree.find(".user-location-button");
        expect(btn.text()).toBe("1 Test Long Loca...");
    });
});
