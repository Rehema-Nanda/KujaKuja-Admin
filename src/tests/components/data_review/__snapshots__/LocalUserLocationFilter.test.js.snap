// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalUserLocationFilter component renders correctly 1`] = `
"<LocalUserLocationFilter locations={{...}} handleUserLocationChange={[Function: mockConstructor]} selectedUserLocations={{...}}>
  <div className=\\"user-location-container\\">
    <div id=\\"UserLocationDropdown\\" className=\\"map-controls-button user-location-button\\" onClick={[Function]}>
      <Badge color=\\"secondary\\" pill={true} tag=\\"span\\">
        <span className=\\"badge badge-secondary badge-pill\\">
          All
        </span>
      </Badge>
       
      Team Location
      <div className=\\"map-controls-button-caret\\">
        <FontAwesomeIcon icon=\\"caret-down\\">
          <i className=\\"fa\\" />
        </FontAwesomeIcon>
      </div>
    </div>
    <mockConstructor placement=\\"bottom\\" isOpen={false} target=\\"UserLocationDropdown\\" toggle={[Function]} className=\\"map-controls-popover\\">
      Popover
    </mockConstructor>
  </div>
</LocalUserLocationFilter>"
`;
